boards/arm/nrf9160_innblue22/innblue22_board_init.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2020 InnBlue
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <init.h>
+#include <drivers/gpio.h>
+
+#define VDD_5V0_PWR_CTRL_GPIO_PIN  21   /* ENABLE_5V0_BOOST  --> speed sensor */
+
+static int pwr_ctrl_init(struct device *dev)
+{
+	struct device *gpio;
+	int    err = -ENODEV;
+
+	/* Get handle of the GPIO device. */
+	gpio = device_get_binding(DT_GPIO_P0_DEV_NAME);
+
+	/* Valid handle? */
+	if (gpio != NULL) {
+
+		/* Configure this pin as output. */
+		err = gpio_pin_configure(gpio, VDD_5V0_PWR_CTRL_GPIO_PIN,
+							GPIO_OUTPUT_ACTIVE);
+		if (err == 0) {
+
+			/* Write "1" to this pin. */
+			err = gpio_pin_set(gpio, VDD_5V0_PWR_CTRL_GPIO_PIN, 1);
+		}
+
+		/* Wait for the rail to come up and stabilize. */
+		k_sleep(K_MSEC(10));
+	}
+
+	/* Operation status? */
+	return (err);
+}
+
+DEVICE_INIT(vdd_5v0_pwr_ctrl_init, "", pwr_ctrl_init, NULL, NULL,
```

Comment:
good question... got a hint on slack and used this in this form (copied from one of nordic boards init)...

for me was "device" as "board" init...
what are the benefits to use sys_init? and what the main differences? 
Will it behave same in "secure" and "non secure" as well?
Thank you in advance!

Reply:
Both ways it will be effectively the same, as both these macros call `DEVICE_AND_API_INIT()` underneath. The thing is that a call to `DEVICE_INIT()` with an empty device name looks a bit artificial, and `SYS_INIT()` is a dedicated macro for such initialization routines.